cadc-test-vos/src/main/java/ca/nrc/cadc/conformance/vos/SyncPullFromVOSpacePackageTest.java

```diff
@@ -0,0 +1,180 @@
+import ca.nrc.cadc.uws.JobReader;
+import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.NodeProperty;
+import ca.nrc.cadc.vos.NodeWriter;
+import ca.nrc.cadc.vos.Protocol;
+import ca.nrc.cadc.vos.Transfer;
+import ca.nrc.cadc.vos.VOS;
+import ca.nrc.cadc.vos.View;
+import com.meterware.httpunit.WebResponse;
+import java.io.StringReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test case for reading data from a service (pullFromVoSpace).
+ *
+ * @author jeevesh
+ */
+public class SyncPullFromVOSpacePackageTest extends VOSTransferTest
+{
+    private static Logger log = Logger.getLogger(SyncPullFromVOSpacePackageTest.class);
+
+    static
+    {
+        Log4jInit.setLevel("ca.nrc.cadc.conformance.vos", Level.INFO);
+    }
+
+    public SyncPullFromVOSpacePackageTest()
+    {
+        super(Standards.VOSPACE_SYNC_21, Standards.VOSPACE_NODES_20);
+    }
+
+    @Test
+    public void testPullFromVOSpacePackage()
+    {
+        try
+        {
+            log.debug("testPullFromVOSpacePackage");
+
+            // Get a DataNode.
+            TestNode dataNode = getSampleDataNode();
+            dataNode.sampleNode.getProperties().add(new NodeProperty(VOS.PROPERTY_URI_CONTENTLENGTH, new Long(1024).toString()));
+            WebResponse response = put(getNodeStandardID(),dataNode.sampleNode, new NodeWriter());
+            assertEquals("PUT response code should be 200", 200, response.getResponseCode());
+
+            // Create the Transfer.
+            List<Protocol> protocols = new ArrayList<Protocol>();
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTP_GET));
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTPS_GET));
+            Transfer transfer = new Transfer(dataNode.sampleNode.getUri().getURI(), Direction.pullFromVoSpace);
+            transfer.getProtocols().addAll(protocols);
+            // Add package view for tar file
+            View packageView = new View(new URI(VOS.VIEW_PACKAGE));
+            packageView.getParameters().add(new View.Parameter(new URI(VOS.PROPERTY_URI_ACCEPT), "tar"));
+            transfer.setView(packageView);
+
+            // Start the transfer.
+            TransferResult result = doSyncTransfer(transfer);
+
+            assertEquals("direction", Direction.pullFromVoSpace, result.transfer.getDirection());
+            assertNotNull("protocols", result.transfer.getProtocols());
+
+            // Get the Job.
+            response = get(result.location);
+            assertEquals("GET of Job response code should be 200", 200, response.getResponseCode());
+
+            // Job phase should be EXECUTING.
+            JobReader jobReader = new JobReader();
+            Job job = jobReader.read(new StringReader(response.getText()));
+            assertEquals("Job phase should be EXECUTING", ExecutionPhase.PENDING, job.getExecutionPhase());
```

Comment:
The expected PHASE should be PENDING because TransferRunner should set it back so that /vault/pkg can run.

Reply:
This section got reworked as well - has been tested with the uws server 1.2.10 (which supports putting job phase back to pending.)